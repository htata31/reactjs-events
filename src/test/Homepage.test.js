import enzyme, {shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import toJson from 'enzyme-to-json';
import HomePage from '../components/Hompage'

enzyme.configure({ adapter: new enzymeAdapterReact16(), });

const contacts_data=[
    {
    name: "Leanne Graham",
    email: "Sincere@april.biz",
    company: {
        name: "Romaguera-Crona",
        catchPhrase: "Multi-layered client-server neural-net",
        }
    }
  ]

function getHomePageShallowWrapper() {
    return shallow(
      <HomePage contacts={contacts_data} />
    );
  }

  describe('The HomePage component', () => {
    test('Taking a snapshot of the homepage', () => {
      const homePageWrapper = getHomePageShallowWrapper();
  
      expect(toJson(homePageWrapper.find('#Contact-info'))).toMatchSnapshot();
    });
  
    
    test('renders the contact name', () => {
      const homePageWrapper = getHomePageShallowWrapper();
      const contactName = homePageWrapper.find('#contact-name');

      expect(contactName.text()).toEqual("Leanne Graham");
    });

    test('renders the contact email', () => {
      const homePageWrapper = getHomePageShallowWrapper();
      const contactEmail = homePageWrapper.find('#contact-email');

      expect(contactEmail.text()).toEqual("Sincere@april.biz");
    });

    test('renders the contact company phrase', () => {
      const homePageWrapper = getHomePageShallowWrapper();
      const contactCompanyPhrase = homePageWrapper.find('#contact-company-phrase');

      expect(contactCompanyPhrase.text()).toEqual("Multi-layered client-server neural-net ");
    });
  });