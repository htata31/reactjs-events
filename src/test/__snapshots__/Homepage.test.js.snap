// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The HomePage component Taking a snapshot of the homepage 1`] = `
<div
  id="Contact-info"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
        id="contact-name"
      >
        Leanne Graham
      </h5>
      <h6
        className="card-subtitle mb-2 text-muted"
        id="contact-email"
      >
        Sincere@april.biz
      </h6>
      <p
        className="card-text"
        id="contact-company-phrase"
      >
        Multi-layered client-server neural-net
         
      </p>
    </div>
  </div>
</div>
`;
